### ISSUE TITLE
Sometimes crashes when trying to save script

### ISSUE THREAD

**Issue by andrisz** [autokey/autokey#285]
## Classification:

Crash

## Reproducibility:

Sometimes. Seemingly random crashes, haven't found a pattern yet.

## Version

AutoKey version: 
0.95.7
Qt version
Installed with debian packages from github (https://github.com/autokey/autokey/releases).
Debian 9.9
Python version: 3.5.3 (default, Sep 27 2018, 17:25:39)  [GCC 6.3.0 20170516]

## Summary

Sometimes AutoKey crashes when I'm trying to save it with ctrl+s after editing script. Haven't found a pattern yet. Can't repeat when I try, but then it crashes when I'm really trying to do something.

## Steps to Reproduce (if applicable)

- edit script
- press ctrl+s

## Expected Results

- script is saved

## Actual Results

- AutoHotkey crashes, I believe this is the relevant log part

```
2019-05-21 21:45:48,995 DEBUG - service - Raw key: 's', modifiers: [<Key.CONTROL: '<ctrl>'>], Key: s
2019-05-21 21:45:48,995 DEBUG - root.Qt-GUI.System-tray-notifier - Show tray icon enabled in settings: True
2019-05-21 21:45:48,997 DEBUG - service - Window visible title: 'AutoKey', Window class: 'autokey-qt.autokey-qt'
2019-05-21 21:45:48,997 INFO - root.Qt-GUI.System-tray-notifier - Rebuilding model item actions, adding all items marked for access through the tray icon.
2019-05-21 21:45:48,998 DEBUG - phrase-menu - Sorting phrase menu by usage count
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/autokey/qtui/centralwidget.py", line 402, in on_save
    item = self.treeWidget.selectedItems()[0]
IndexError: list index out of range
Fatal Python error: Aborted
```

**Comment by andrisz** [comment]
Okay, so when you click on the empty part in the script tree, the selected script deselects in tree (it still stays opened in the right part) and when you try saving it - crash. Now it seems apparent from the crash log :/

**Comment by josephj11** [comment]
This sounds like a real bug that needs to be addressed, but ...

You probably know this, but once a script has been saved, you can edit it directly using your favorite text editor. The one built into AutoKey works (when you can find the cursor), but it's far from ideal.

You can leave your scripts in the directory tree that AutoKey creates ($HOME/.config/autokey/data) or you can create an external directory using the AutoKey main window and put some or all of your scripts and phrases anywhere you like to make them more convenient to access.

**Comment by andrisz** [comment]
Thanks, didn't know, just started using AutoKey yesterday. So AutoKey will detect updates in scripts and automatically reload them? Great, that means I can go back to gtk version (started using qt bcause gtk editor font was too small).

**Comment by luziferius** [comment]
Hmm. That stack trace indicates that no item is selected during the save. So you are right. Thank you for pointing out this issue.

This is a rather dumb mistake. The save logic should not depend on the selection in that list. Rather, it should read the to-be saved item right from the editor and not bother asking the tree view.

**Comment by luziferius** [comment]
@andrisz Fixed in [v0.95.8](https://github.com/autokey/autokey/releases/tag/v0.95.8). I know, that took a while…
